### PR TITLE
mgr/dashboard: config options table cleanup

### PR DESCRIPTION
**Pull request by s0nea**
Remove columns 'tags', 'enum_values', 'long_desc', 'type', 'flags', 'daemon_default', 'desc', 'level', 'can_update_at_runtime', 'services', 'max', 'see_also', 'min' and 'source' from table view and add them to the details.
The table contains 'name', 'value' and 'default' only.

config_options_cleanup

Fixes: http://tracker.ceph.com/issues/34533

Signed-off-by: Tatjana Dehler <tdehler@suse.com>

- [x] References tracker ticket

### CODE REVIEW

**Comment by votdev** [comment]
Like the new look.

**Comment by votdev** [comment]
What about using badges for the flags and services?

auswahl_005

```
Index: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-details/configuration-details.component.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-details/configuration-details.component.html	(date 1539180752000)
+++ src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-details/configuration-details.component.html	(date 1539248243000)
@@ -56,9 +56,9 @@
           <td i18n
               class="bold col-sm-1">Flags</td>
           <td class="col-sm-3">
-            <span *ngFor="let flag of selectedItem.flags; last as isLast">
+            <span *ngFor="let flag of selectedItem.flags">
               <span title="{{ flags[flag] }}">
-                {{ flag | uppercase }}{{ !isLast ? "," : "" }}<br/>
+                <span class="badge badge-pill badge-primary margin-right-sm">{{ flag | uppercase }}</span>
               </span>
             </span>
           </td>
@@ -66,7 +66,11 @@
         <tr>
           <td i18n
               class="bold col-sm-1">Services</td>
-          <td class="col-sm-3">{{ selectedItem.services }}</td>
+          <td class="col-sm-3">
+            <span *ngFor="let service of selectedItem.services">
+              <span class="badge badge-pill badge-primary margin-right-sm">{{ service }}</span>
+            </span>
+          </td>
         </tr>
         <tr>
           <td i18n
Index: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-details/configuration-details.component.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-details/configuration-details.component.ts	(date 1539180752000)
+++ src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-details/configuration-details.component.ts	(date 1539248243000)
@@ -1,5 +1,7 @@
 import { Component, Input, OnChanges } from '@angular/core';
 
+import * as _ from 'lodash';
+
 import { CdTableSelection } from '../../../../shared/models/cd-table-selection';
 
 @Component({
@@ -27,6 +29,7 @@
   ngOnChanges() {
     if (this.selection.hasSelection) {
       this.selectedItem = this.selection.first();
+      this.selectedItem.services = _.split(this.selectedItem.services, ',');
     }
   }
 }
```

**Comment by s0nea** [comment]
> What about using badges for the flags and services?

Good idea, I agree!

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.ts [hunk 1]
```diff
@@ -110,25 +100,7 @@ export class ConfigurationComponent implements OnInit {
...
-      { flexGrow: 1, prop: 'type' },
-      { flexGrow: 1, prop: 'level' },
-      { flexGrow: 1, prop: 'default', cellClass: 'wrap' },
-      { flexGrow: 2, prop: 'daemon_default', name: 'Daemon default' },
-      { flexGrow: 1, prop: 'tags', name: 'Tags' },
-      { flexGrow: 1, prop: 'services', name: 'Services' },
-      { flexGrow: 1, prop: 'see_also', name: 'See_also', cellClass: 'wrap' },
-      { flexGrow: 1, prop: 'max', name: 'Max' },
-      { flexGrow: 1, prop: 'min', name: 'Min' }
+      { flexGrow: 1, prop: 'default', cellClass: 'wrap' }
```
Comment (ricardoasmarques) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.ts]: IMO, both `default` and `current value`  columns should have the same `flexGrow` (or you can simply remove the `flexGrow` from all columns).

**Comment by tspmelo** [comment]
Could we keep the description column on the table? I think it's a useful information to read when going through all the settings.

**Comment by s0nea** [comment]
@tspmelo:
@LenzGr suggested to only have 'Name', 'Current value' and 'Default on the table: http://tracker.ceph.com/issues/34533, so I think we should discuss it with him next week.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.ts (hunk 1 above)
Comment (s0nea) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.ts]: Okay, will do.

**Comment by LenzGr** [comment]
> Could we keep the description column on the table? I think it's a useful information to read when going through all the settings.

No objections - as long as it does not make the table look too crowded/cluttered?

**Comment by s0nea** [comment]
> No objections - as long as it does not make the table look too crowded/cluttered?

I don't think so. I will update the PR and add a screenshot.

**Comment by s0nea** [comment]
screenshot_2018-10-15_11-26-56

I think it's fine to keep the description column. What's you opinion?

**Comment by s0nea** [comment]
jenkins test make check arm64

**Pull request merged**